connect-inject/container_init.go

```diff
@@ -11,11 +11,15 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 )
 
+// The init container is bounded in memory usage by the size of the consul binary,
```

Comment:
Can you go into more detail here about how the init container runs `cp` to copy the consul binary to a shared volume and that this may copy the whole binary into memory which means our memory needs to be > the size of the consul binary.

Also that we're only setting the limit because we didn't want the request to be too large.